contracts/EventStorage.sol

```diff
@@ -28,40 +32,51 @@ contract EventStorage {
 
     constructor (string memory _version) {
         version = _version;
+        INSULINS.push(Insulin('Novorapid', 10800 ));
+        INSULINS.push(Insulin('Humuline',  18000 ));
     }
 
     function bolus (
         uint256 _timestamp,          //Unix timestamp
         uint256 _glucose,       //Current blood glucose reading
         uint256 _carbs, //Estimated carbohydrate intake
         uint256 _units,     //Units of insulin the app has calculated for you
-        Insulin calldata _insulinType // lookup table with insulin name (string) and duration in minutes 
+        uint256 _insulinIndex // index from lookup table with insulin name (string) and duration in seconds 
          
     ) 
         external 
     {
         require(_timestamp > 0, "Cant be zero");
         //TODO
         //add more checks if need like above;
-
-        emit MeasureRecord (
-            msg.sender,
-            _timestamp,
-            _glucose,
-            _carbs,
-            _units,
-            _insulinType
-        );
-
+        if  (personalInsulins[msg.sender].length == 0) {
```

Comment:
What if I have added at least one personal insulin? I get the impression the hardcoded (built-in) insulins are removed from view.

Reply:
Yes you are right. We can use only one of two arrays: common harcoded or personal